.github/workflows/schedule-posts.txt

```diff
@@ -0,0 +1,40 @@
+name: Publish posts scheduled for today
+
+on:
+  schedule:
+    # Run every day at 23:59 UTC or manually run
+    - cron: "30 23 * * *"
+  workflow_dispatch:
+
+jobs:
+  publish_drafts:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: master
+
+      - name: Get the date for today
+        id: date
+        run: echo "TODAY=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+      - name: Check for drafts and move to posts
+        run: |
+          today=$(date +'%Y-%m-%d')
```

Comment:
Since you added this here, the step before became redundant. You should merge it with this step since `TODAY` is used in the last step and query for the date only once